napalm_eos/eos.py

```diff
@@ -1473,3 +1473,66 @@ def get_config(self, retrieve="all"):
             }
         else:
             raise Exception("Wrong retrieve filter: {}".format(retrieve))
+
+
+    def ping(self, destination, source='', ttl=255, timeout=2, size=100, count=5):
+        """
+        Execute ping on the device and returns a dictionary with the result.
+        Output dictionary has one of following keys:
+            * success
+            * error
+        In case of success, inner dictionary will have the followin keys:
+            * probes_sent (int)
+            * packet_loss (int)
+            * rtt_min (float)
+            * rtt_max (float)
+            * rtt_avg (float)
+            * rtt_stddev (float)
+            * results (list)
+        'results' is a list of dictionaries with the following keys:
+            * ip_address (str)
+            * rtt (float)
+        """
+        ping_dict = {}
+        command = 'ping {}'.format(destination)
+        command += ' timeout {}'.format(timeout)
+        command += ' size {}'.format(size)
+        command += ' repeat {}'.format(count)
+        if source != '':
+            command += ' source {}'.format(source)
+        output = self.device.run_commands(command, encoding='text')[0]['output']
```

Comment:
I think this line should be:

    output = self.device.run_commands([command], encoding='text')[0]['output']